frontend/src/metabase/query_builder/selectors.js

```diff
@@ -914,6 +919,28 @@ export const getNativeEditorSelectedText = createSelector(
   },
 );
 
+export const getAllNativeEditorSelectedText = createSelector(
+  [getNativeEditorSelectedRanges, getNextRunDatasetQuery],
+  (selectedRanges, query) => {
+    if (
+      selectedRanges == null ||
+      selectedRanges.length === 0 ||
+      query == null ||
+      query.native == null
+    ) {
+      return null;
+    }
+    const queryText = query.native.query;
+    const selectedText = selectedRanges.map((range) =>
+      queryText.slice(
+        getOffsetForQueryAndPosition(queryText, range.start),
+        getOffsetForQueryAndPosition(queryText, range.end),
+      ),
+    );
+    return selectedText.join(" ");
```

Comment:
nit: unexpected whitespace insertion when only a part of a string is selected

![image](https://github.com/user-attachments/assets/9d708390-2f75-4658-9a9a-799588b1eadc)

This will result in `select "foo bar"` instead of `select "foobar"`.

It's probably acceptable though. Feel free to ignore.